services/frontend/src/app/AuthenticationLayer.tsx

```diff
@@ -16,21 +16,29 @@ const AuthenticationLayer = ({ children }: Props) => {
   const dispatch = useDispatch();
   const router = useRouter();
 
+  const fetchSessionToken = async () => {
+    const response = await fetch("http://localhost:3000/api/me");
+    return response.json();
+  };
+
   useEffect(() => {
-    if (session) {
-      dispatch(
-        sessionSignIn({
-          currentUser: session.currentUser,
-          image: session.user?.image || null,
-          sessionToken: null,
-          isLoggedIn: true,
-        }),
-      );
-
-      if (session.currentUser == null) {
-        router.push("/onboarding");
+    const getUserInfo = async () => {
+      if (session) {
+        dispatch(
+          sessionSignIn({
+            currentUser: session.currentUser,
+            image: session.user?.image || null,
+            sessionToken: await fetchSessionToken(),
```

Comment:
Since this function can potentially throw an error you should do proper error handling before dispatching the resulf of the function

e.g.

```
  const fetchSessionToken = async () => {
      const response = await fetch('/api/me'); // Using a relative URL here
      return await response.json();
  };

  const getUserInfo = async () => {
    if (session) {
      try {
        const sessionToken = await fetchSessionToken();
        dispatch(
          sessionSignIn({
            currentUser: session.currentUser,
            image: session.user?.image || null,
            sessionToken: sessionToken,
            isLoggedIn: true,
          }),
        );

        if (session.currentUser == null) {
          router.push('/onboarding');
        }
      } catch (error) {
        // dispatch an error notification
      }
    }
  };

  useEffect(() => {
    getUserInfo();
  }, [session, dispatch, router]);
```